### PR TITLE
Account service adjust

### PR DESCRIPTION
**Pull request by NiazBikbaev**

### CODE REVIEW

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java [hunk 1]
```diff
@@ -16,24 +20,68 @@
...
+        return exists;
     }
 
     @Override
     public Account getAccountByUsername(String username) {
-        return accountRepository.findByUsername(username);
+        Account account;
+        if (username == null) {
+            throw new ServiceException("Username can't be null");
+        }
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: You can replace your null checks with  Lomboks `@NonNull` annotation right before the parameter in methoid
You can read about it [here](https://projectlombok.org/features/NonNull)
There is really no need to wrap such exceptions in some kinda custom. 
At this project it is up to you, will you use `@NonNull` or will you write it explictly, just be aware of `@NonNull` annotation of Lombok

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java [hunk 2]
```diff
@@ -16,24 +20,68 @@
...
     @Override
     public Account registerAccount(Account account) {
-        Account savedAccount = null;
-        savedAccount = accountRepository.save(account);
+        Account savedAccount;
+        try {
+            savedAccount = accountRepository.save(account);
+        } catch (DataIntegrityViolationException e) {
+            log.error(e.getMessage(), e);
+            throw new ServiceException("Could not register account: account with such username already exists", e);
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: Message

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java [hunk 3]
```diff
@@ -16,24 +20,68 @@
...
+        }
+        return account;
     }
 
     @Override
     public boolean isUserNameExists(String username) {
-        return accountRepository.existsByUsername(username);
+        boolean exists;
+        if (username == null) {
+            throw new ServiceException("Username can't be null");
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: Message

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java [hunk 4]
```diff
@@ -16,24 +20,68 @@
...
+            throw new ServiceException("Database error", e);
+        }
+        return exists;
     }
 
     @Override
     public Account getAccountByUsername(String username) {
-        return accountRepository.findByUsername(username);
+        Account account;
+        if (username == null) {
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: Objects.isNull()

File: src/test/java/com/epam/magazinestore/AccountServiceTests.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Test
+    public void registerNonUniqueAccountTest() {
+        expectedException.expect(ServiceException.class);
+        expectedException.expectMessage("Could not register account: account with such username already exists");
+        Account account = new Account();
+        account.setUsername("user");
+        Account anotherAccount = new Account();
+        anotherAccount.setUsername("user");
```
Comment (AntonKiselyov) [inline src/test/java/com/epam/magazinestore/AccountServiceTests.java]: Test constant

**Pull request merged**